NasdaqDataLink.cs

```diff
@@ -0,0 +1,194 @@
+        protected NasdaqDataLink(string valueColumnName)
+        {
+            _valueColumn = valueColumnName;
+        }
+
+        /// <summary>
+        /// Using the Nasdaq Data Link V3 API automatically set the URL for the dataset.
+        /// </summary>
+        /// <param name="config">Subscription configuration object</param>
+        /// <param name="date">Date of the data file we're looking for</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>STRING API Url for Nasdaq Data Link.</returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            var source = $"https://data.nasdaq.com/api/v3/datasets/{config.Symbol.Value}.csv?order=asc&api_key={_authCode}";
+            return new SubscriptionDataSource(source, SubscriptionTransportMedium.RemoteFile);
+        }
+
+        /// <summary>
+        /// Parses the data from the line provided and loads it into LEAN
+        /// </summary>
+        /// <param name="config">Subscription configuration</param>
+        /// <param name="line">Line of data</param>
+        /// <param name="date">Date</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>New instance</returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            // be sure to instantiate the correct type
+            var data = (NasdaqDataLink) Activator.CreateInstance(GetType());
+            data.Symbol = config.Symbol;
+            var csv = line.Split(',');
+
+            if (!_isInitialized)
+            {
+                _isInitialized = true;
+                foreach (var propertyName in csv)
+                {
+                    var property = propertyName.Trim().ToLowerInvariant();
+                    data.SetProperty(property, 0m);
+                    _propertyNames.Add(property);
+                }
+                // Returns null at this point where we are only reading the properties names
+                return null;
+            }
+
+            data.Time = DateTime.ParseExact(csv[0], "yyyy-MM-dd", CultureInfo.InvariantCulture);
+            data.EndTime = data.Time + TimeSpan.FromDays(1);
+
+            for (var i = 1; i < csv.Length; i++)
+            {
+                var value = csv[i].ToDecimal();
+                data.SetProperty(_propertyNames[i], value);
+            }
+            
+            // Check if the data is price data
+            var priceKey = new List<string> {"close", "price", "settle", "value"};
+            var commonList = _propertyNames.Intersect(priceKey).ToList();
+            
+            if (commonList.Any())
+            {
+                // If so, set .Value as the first common element with price keyword
+                data.Value = (decimal)data.GetProperty(commonList[0]);
```

Comment:
Cool! we should still allow users to overwrite/specify this somehow I think, would it be possible? nice unit test to add 👍 

`_valueColumn` is no longer being used BTW, maybe it's optional now, default value null and we try `{"close", "price", "settle", "value"}` and if desired user can specify it